vault/expiration.go

```diff
@@ -114,6 +119,91 @@ func (c *Core) stopExpiration() error {
 	return nil
 }
 
+// Tidy cleans up stale leases which are associated with invalid tokens
+func (m *ExpirationManager) Tidy() error {
+	var tidyErrors *multierror.Error
+
+	// Create a cache to keep track of looked up tokens
+	tokenCache := make(map[string]string)
```

Comment:
The hit for an invalid entry isn't as high, because there won't be any value to decode/de-serialize. But memoizing both statues is fine -- however, you don't need a string, you just need a bool. Presence of the entry in the map means it's been seen; the bool indicates whether it was valid or not.